app/decorators/user_decorator.rb

```diff
@@ -58,6 +58,21 @@ def active_or_pending_profile
     user.active_profile || pending_profile
   end
 
+  def pending_profile_requires_verification?
```

Comment:
wow! the logic in here is sufficiently complicated that I wonder if we should consider breaking out a separate class. If not now, maybe l8r.